### ISSUE TITLE
No Burmy, Sewaddle, or Galarian Corsola Icons

### ISSUE THREAD

**Issue by Nightlight72** [Emery319/E19-CobblemonMinimapIcons#10]
its the only minimap resourcepack that i have on and these two for some reason just dont have icons (ive also only seen grass cloak burmy in the server i play on, idk if the other cloaks have icons; this also goes for wormadam)

**Comment by Emery319** [comment]
Burmy and Wormadam have not yet been included in the official cobblemon mod, so this is most likely a Pokémon added by the server with an addon, do you have any clue as to what addon the server used, I'd need the file format of the addon to fix this.

**Comment by Nightlight72** [comment]
trying to ask, ill let you know when i get an answer

**Comment by Nightlight72** [comment]
so burmy is from the pocketmonsters datapack and sewaddle is from wavimons, but someone says pocketmonsters is taken down for some reason https://modrinth.com/datapack/wavimons-(cobblemon)

edit: found out cursola also has a missing icon so i would assume is the same for galarian corsola; this one is from missingmons https://modrinth.com/datapack/missingmons-cobblemon

**Comment by Emery319** [comment]
Sorry for the delay in handling this. The reason Cursola was displaying incorrectly is that the texture file name was not cursola.png but rather cursola_solid.png. After modifying it, it now displays correctly.

As for Sewaddle in Wavimons, it displayed correctly during my testing, so it might be due to a conflict with another addon. If you find the cause of the issue, feel free to reach out to me anytime.

It’s true that PocketMönsters can no longer be downloaded, but the creator’s server is still using the texture pack (they’ve corrupted the server resource pack’s file header to prevent extraction). The issue is that the texture for Burmy with Plant Cloak doesn’t have the _plant suffix; it’s named burmy.png. After modifying it, both the Plant Cloak Burmy and the Wormadam display correctly.

Just download this [pokemon.json](https://github.com/user-attachments/files/17412643/pokemon.json) file and replace it in your resource pack under the path assets/xaerominimap/entity/icon/definition/cobblemon, and it should work!

**Comment by Nightlight72** [comment]
alright, thank you and have a good day

**Comment by Nightlight72** [comment]
so i found another one that doesnt have icons lmao; shellos is from missingmons

edit: galarian corsola still doesnt display the icon for some reason but im gonna assume thats another conflict

**Comment by Emery319** [comment]
[pokemon.json](https://github.com/user-attachments/files/17430554/pokemon.json) <- I have fixed the display issues for Morpeko, Lycanroc, Palafin, Shaymin, Shellos, Gastrodon, and Ogerpon. They should now be compatible with MissingMons.

**Comment by Nightlight72** [comment]
i am sorry that i keep finding mons without icons lmao; hoppip is from wavimons, and if it doesnt work after being fixed in the resource pack ima just assume wavimons has conflict with smth else in the server i play in

**Comment by Emery319** [comment]
Using WaviMon alone is fine, so it must be other addons causing the problem.

**Comment by Nightlight72** [comment]
fair enough, but have you looked to see if hoppip line is fixed?

**Comment by Emery319** [comment]
![2024-10-19_08 41 47](https://github.com/user-attachments/assets/7f640add-a650-4172-939f-67cb9990596c)
![2024-10-19_08 45 02](https://github.com/user-attachments/assets/fd594375-a804-4758-a8f6-45336c1b453b)

You can try it yourself by downloading a WaviMon, I checked it yesterday when I was fixing MissingMons and it didn't turn out to be a problem.

**Comment by Nightlight72** [comment]
alr im just gonna assume its smth with the server then, thanks

**Comment by Nightlight72** [comment]
unown also doesnt have its icons, alphabet soup https://modrinth.com/datapack/alphabetsoup (i am so sorry for how many i have been posting lmao)

**Comment by Emery319** [comment]
The issue with Unown requires modifications to the “resolvers” file. Otherwise, all Unown forms will display as the same letter. This change is fairly time-consuming, and once Cobblemon officially implements Unown, resource packs that have modified “resolvers” will crash and become unusable. Additionally, I would need to redo all of this work from scratch. Because of this, I don’t plan to working on it. 

Here’s a modified “[pokemon.json](https://github.com/user-attachments/files/17445302/pokemon.json)” file where all Unown will be displayed as the letter “A”. If that’s acceptable, you can use it for now as a temporary solution.

Currently, I’m busy with real-life commitments and preparing for the Cobblemon 1.6 update. As a result, I might not have plans to work on adapting addons for now. I hope you understand.

**Comment by Nightlight72** [comment]
yeah i understand, thats alright. i appreciate the changes youve done so far

**Comment by Emery319** [comment]
Also, you can still post conflicting addons here. I’ll deal with them after I’m done with the current stuff!

**Comment by Nightlight72** [comment]
hoopa doesnt have an icon; its from pokemans pack https://modrinth.com/datapack/cobblemon-pokemans

**Comment by Nightlight72** [comment]
hisuian growlithe doesnt have icon; its from genomons https://modrinth.com/datapack/genomons

**Comment by Emery319** [comment]
It should now fully support Pokemans Pack, GenoMons, and Ascension Megamons. This time, in addition to replacing the “[pokemon.json](https://github.com/user-attachments/files/17542696/pokemon.json)” file, you’ll also need to add 5 Pokémon icons each to the “regular” and “shiny” folders under the path assets/xaerominimap/entity/icon/sprite. These icons are included in the “[sprite.zip](https://github.com/user-attachments/files/17542715/sprite.zip)” file. However, it seems that the “Armored Mewtwo” added by GenoMons is still incomplete, as using a command to summon it will cause the game to crash.

**Comment by Nightlight72** [comment]
neat, also for some reason girafarig appeared as black and pink boxes on my minimap (its from pocketmonsters)

**Comment by Emery319** [comment]
Girafarig includes gender differences, and it seems that an add-on has added a female variant. However, the missing texture issue arises because the resource pack lacks the necessary icons. Using this updated "[sprite.zip](https://github.com/user-attachments/files/17580934/sprite.zip)" file should resolve the problem.

**Comment by Nightlight72** [comment]
alr, ty

**Comment by Nightlight72** [comment]
solosis and indeedee dont have icons; wavimons

**Comment by Emery319** [comment]
My testing shows that Solosis displays correctly in WaviMon, and Indeedee is not included in this add-on. So, this is likely a conflict caused by another add-on, similar to the previous issue with the Hoppip and Sewaddle evolution lines.

**Comment by Nightlight72** [comment]
the tapus dont have icons; mysticmons https://modrinth.com/datapack/mysticmons

**Comment by Emery319** [comment]
Here are the updated "[pokemon.json](https://github.com/user-attachments/files/17616267/pokemon.json)" and "[sprite.zip](https://github.com/user-attachments/files/17616270/sprite.zip)", with fixes for Tapufini, Giratina, Meloetta, and Necrozma. It should now fully support MysticMons.

**Comment by Nightlight72** [comment]
alr; galarian yamask line doesnt have icon, hiddenmons https://modrinth.com/datapack/hiddenmons-cobblemon

**Comment by Emery319** [comment]
"[pokemon.json](https://github.com/user-attachments/files/17626091/pokemon.json)" & "[sprite.zip](https://github.com/user-attachments/files/17626092/sprite.zip)" Done, with fixes for Minior, Runerigus, and Zygarde; HiddenMons support has been added.

**Comment by Nightlight72** [comment]
zygarde doesnt have icon; hiddenmons https://modrinth.com/datapack/hiddenmons-cobblemon/gallery

**Comment by Emery319** [comment]
Zygarde appears normal on my end. The issue might be caused by your server modifying the file names for HiddenMons or a conflict with other addons. I have updated the resource pack to version 1.2.7. If the problem persists after updating, you can send me the resource pack used on your server, and I will adapt it accordingly.

**Comment by Nightlight72** [comment]
ok i downloaded the new version and that seemed to fix it, ty lol
